docs/getting-started.md

```diff
@@ -0,0 +1,395 @@
+# Contributing to OSTree
+
+The following guide is about OSTree forking, building, adding a command, testing the command, and submitting the change.
+
+- [Getting Started](#getting-started)
+- [Building OSTree](#building-ostree)
+    - [Install Build Dependencies](#install-build-dependencies)
+    - [OSTree Build Commands](#ostree-build-commands)
+- [Testing a Build](#testing-a-build)
+    - [Testing in a Container](#testing-in-a-container)
+    - [Testing in a Virtual Machine](#testing-in-a-virtual-machine)
+- [Tutorial: Adding a basic builtin command to OSTree](#tutorial-adding-a-basic-builtin-command-to-ostree)
+    - [Modifying OSTree](#modifying-ostree)
+    - [OSTree tests](#ostree-tests)
+    - [Submitting patch](#submitting-a-patch)
+    - [Commit message style](#commit-message-style)
+    - [Returning workflow](#returning-workflow)
+
+---
+
+## [Getting Started](#getting-started)
+
+Fork https://github.com/ostreedev/ostree, the run the following commands.
+
+```bash
+$ git clone https://github.com/<username>/ostree
+$ git remote add upstream https://github.com/ostreedev/ostree
+$ git checkout master
+$ git branch --set-upstream-to=upstream/master master
+```
+Make a branch from master for your patch.
+```bash
+$ git checkout -b <name-of-branch>
+$ git branch --set-upstream-to=upstream/master <name-of-branch>
+```
+
+## [Building OSTree](#building-ostree)
+
+### [Install Build Dependencies](#install-build-dependencies)
+
```

Comment:
I'd put a link in this section to the larger `build-dependencies.md` noting there are additional details around the dependencies in that doc.